contracts/eosio.bios/include/eosio.bios/eosio.bios.hpp

```diff
@@ -78,8 +80,14 @@ namespace eosiobios
     */
    class [[eosio::contract("eosio.bios")]] bios : public eosio::contract
    {
+   private:
+      eosio::symbol system_resource_currency;
```

Comment:
Antelope contracts do not have dynamic private variables. This is because contracts are compiled to WASM programs which execute within the Antelope virtual machine "EOSVM". While executing as a WASM program, they inherently have no persistent storage.

Persistent data storage is handled through a WASM intrinsic API, which is exposed through the C++ multi_index_table API:
https://docs.eosnetwork.com/docs/latest/smart-contracts/state-data
https://docs.antelope.io/cdt/latest/how-to-guides/multi-index/how-to-define-a-primary-index

There are three strategies for having the `system_resource_currency`:
1. Use a static constant which is hardcoded into the contract. something like `eosio::symbol system_resource_currency; https://stackoverflow.com/questions/177437/what-does-const-static-mean-in-c-and-c
2. Use the same variable, set it during the constructor of the contract https://docs.antelope.io/cdt/latest/reference/Classes/classeosio_1_1contract#function-contract
3. Use a dynamic variable that uses multi_index_table API to set/get the currency name. As we will always use the same currency contract, this is less efficient and more complex. I suggest we do not do this.

Suggest use strategy 1. You can then delete the `syscurrency()` function